processor/metricsgenerationprocessor/processor_test.go

```diff
@@ -392,130 +393,173 @@ func getOutputForIntGaugeTest() pmetric.Metrics {
 		},
 		{
-			name:    "metric2_zero_subtract",
-			testDir: "metric2_zero_value",
+			name:                       "metric2_zero_add",
+			testDir:                    "metric2_zero_value",
+			matchAttributesFlagEnabled: true,
 		},
 		{
-			name:    "metric2_zero_multiply",
-			testDir: "metric2_zero_value",
+			name:                       "metric2_zero_subtract",
+			testDir:                    "metric2_zero_value",
+			matchAttributesFlagEnabled: true,
 		},
 		{
-			name:    "metric2_zero_divide",
-			testDir: "metric2_zero_value",
+			name:                       "metric2_zero_multiply",
+			testDir:                    "metric2_zero_value",
+			matchAttributesFlagEnabled: true,
 		},
 		{
-			name:    "metric2_zero_percent",
-			testDir: "metric2_zero_value",
+			name:                       "metric2_zero_divide",
+			testDir:                    "metric2_zero_value",
+			matchAttributesFlagEnabled: true,
+		},
+		{
+			name:                       "metric2_zero_percent",
+			testDir:                    "metric2_zero_value",
+			matchAttributesFlagEnabled: true,
+		},
+		{
+			name:                       "match_attributes_disabled",
+			testDir:                    "match_attributes",
+			matchAttributesFlagEnabled: false,
+		},
+		{
+			name:                       "match_attributes_enabled",
+			testDir:                    "match_attributes",
+			matchAttributesFlagEnabled: true,
 		},
 	}
 
 	for _, testCase := range testCaseNames {
-		cm, err := confmaptest.LoadConf(filepath.Join("testdata", testCase.testDir, "config.yaml"))
-		assert.NoError(t, err)
-
-		next := new(consumertest.MetricsSink)
-		factory := NewFactory()
-		cfg := factory.CreateDefaultConfig()
-
-		sub, err := cm.Sub(fmt.Sprintf("%s/%s", "experimental_metricsgeneration", testCase.name))
-		require.NoError(t, err)
-		require.NoError(t, sub.Unmarshal(cfg))
-
-		mgp, err := factory.CreateMetrics(
-			context.Background(),
-			processortest.NewNopSettings(),
-			cfg,
-			next,
-		)
-		assert.NotNil(t, mgp)
-		assert.NoError(t, err)
-
-		assert.True(t, mgp.Capabilities().MutatesData)
-		require.NoError(t, mgp.Start(context.Background(), nil))
-
-		inputMetrics, err := golden.ReadMetrics(filepath.Join("testdata", testCase.testDir, "metrics_input.yaml"))
-		assert.NoError(t, err)
-
-		err = mgp.ConsumeMetrics(context.Background(), inputMetrics)
-		assert.NoError(t, err)
-
-		got := next.AllMetrics()
-		// golden.WriteMetrics(t, filepath.Join("testdata", testCase.testDir, fmt.Sprintf("%s_%s", testCase.name, "expected.yaml")), got[0])
-		expected, err := golden.ReadMetrics(filepath.Join("testdata", testCase.testDir, fmt.Sprintf("%s_%s", testCase.name, "expected.yaml")))
-		assert.NoError(t, err)
-		assert.Len(t, got, 1)
-		err = pmetrictest.CompareMetrics(expected, got[0],
-			pmetrictest.IgnoreMetricDataPointsOrder(),
-			pmetrictest.IgnoreStartTimestamp(),
-			pmetrictest.IgnoreTimestamp())
-		assert.NoError(t, err)
+		t.Run(testCase.name, func(t *testing.T) {
```

Comment:
I modified this test's setup so each test case can be run independently locally, and the test name in failures will be more obvious. The actual test functionality here remains largely unchanged, other than setting the feature gate value.